### PR TITLE
Add limited support for OCaml 4.12 into omp1

### PR DESCRIPTION
**Pull request by hhugo**
Reusing part of https://github.com/ocaml-ppx/ocaml-migrate-parsetree/pull/107

### CODE REVIEW

**Comment by ghost** [comment]
The code looks good to me. There just seems to be some CI failures on older versions of OCaml due to references to `Option`.

File: src/migrate_parsetree_412_411_migrate.ml [hunk 1]
```diff
@@ -0,0 +1,1525 @@
...
+        (List.map
+           (fun x ->
+              let (x0, x1) = x in
+              ((copy_out_sig_item x0), (Option.map copy_out_value x1))) x0)
+  | Ast_412.Outcometree.Ophr_exception x0 ->
+      Ast_411.Outcometree.Ophr_exception
+        (let (x0, x1) = x0 in (x0, (copy_out_value x1)))
+
+and copy_out_type_param : Ast_412.Outcometree.out_type_param -> string * (bool * bool) =
+  function (str, (_,_)) -> str, (true,true)
```
Comment (hhugo) [inline src/migrate_parsetree_412_411_migrate.ml]: This need to be fixed

File: src/migrate_parsetree_411_412_migrate.ml [hunk 2]
```diff
@@ -0,0 +1,1519 @@
...
+      Ast_412.Outcometree.Ophr_signature
+        (List.map
+           (fun x ->
+              let (x0, x1) = x in
+              ((copy_out_sig_item x0), (Option.map copy_out_value x1))) x0)
+  | Ast_411.Outcometree.Ophr_exception x0 ->
+      Ast_412.Outcometree.Ophr_exception
+        (let (x0, x1) = x0 in (x0, (copy_out_value x1)))
+and copy_out_type_param : string * (bool * bool) -> Ast_412.Outcometree.out_type_param  =
+  function (str, (_,_)) -> str, (Covariant,Injective)
```
Comment (hhugo) [inline src/migrate_parsetree_411_412_migrate.ml]: This need to be fixed

**Comment by hhugo** [comment]
Ci failure seems to be unrelated

**Comment by hhugo** [comment]
CI failure still seem to be unrelated

**Pull request merged**